test/hotspot/jtreg/serviceability/jvmti/events/NotifyFramePopStressTest/NotifyFramePopStressTest.java

```diff
@@ -0,0 +1,136 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Verifies NotifyFramePop request is cleared if JVMTI_EVENT_FRAME_POP is disabled
+ * @requires vm.jvmti
+ * @library /test/lib
+ * @compile NotifyFramePopStressTest.java
+ * @run main/othervm/native -agentlib:NotifyFramePopStressTest NotifyFramePopStressTest
+ */
+
+import jtreg.SkippedException;
+
+public class NotifyFramePopStressTest {
+    static volatile boolean done = false;
+    static volatile int notifyCount = 0;
+
+    static {
+        try {
+            System.loadLibrary("NotifyFramePopStressTest");
+        } catch (UnsatisfiedLinkError ex) {
+            System.err.println("Could not load NotifyFramePopStressTest library");
+            System.err.println("java.library.path:"
+                + System.getProperty("java.library.path"));
+            throw ex;
+        }
+    }
+
+    public static void main(String args[]) {
+        if (!canGenerateFramePopEvents()) {
+            throw new SkippedException("FramePop event is not supported");
+        }
+        Thread testThread = Thread.currentThread();
+        Thread controlThread = new Thread(() -> control(testThread), "Control Thread");
+
+        setFramePopNotificationMode(testThread, true);
+        controlThread.start();
+        sleep(10);
+
+        for (int i = 0; i < 10*1000; i++) {
+            foo();
+            bar();
+        }
+        done = true;
+
+        try {
+            controlThread.join();
+        } catch (InterruptedException e) {
+        }
+
+        if (failed()) {
+            throw new RuntimeException("Test FAILED: see log for details");
+        } else {
+            log("Test PASSED");
+        }
+    }
+
+    private static void sleep(long ms) {
+        try {
+            Thread.sleep(ms);
+        } catch (InterruptedException e) {
+        }
+    }
+
+    private static void control(Thread thread) {
+        System.out.println("control has started");
+        while (!done) {
+            suspend(thread);
+            if (notifyFramePop(thread)) {
+                notifyCount++;
+                System.out.println("control incremented notifyCount to " + notifyCount);
+            }
+            resume(thread);
+            int waitCount = 0;
+            while (notifyCount != getPopCount()) {
+                sleep(1);
+                waitCount++;
+                if (waitCount > 1000) {
+                    break;
+                }
+            }
+            if (waitCount > 50) {
```

Comment:
It was written to fail if the count is over 50, but it will wait up to 1000 before exiting the loop to provide a bit of debugging info. If it gets to a 1000, then the test is probably stuck. If it fail let's say at 75, then maybe the test just needs an adjustment to wait for more iterations.

Reply:
I've changed it from 50 to 100 but this can be not enough for `-Xcomp`.
I've never seen it yet failed with `-Xcomp` though.